X-Account-Key: account3
X-UIDL: 1169070910.575
X-Mozilla-Status: 0001
X-Mozilla-Status2: 10000000
Return-Path: <fink-devel-bounces@lists.sourceforge.net>
Received: from mwinf5128.me-wanadoo.net (mwinf5128.me-wanadoo.net)
	by mwinb2306 (SMTP Server) with LMTP; Mon, 22 Jan 2007 00:05:11 +0100
X-Sieve: Server Sieve 2.2
Received: from me-wanadoo.net (localhost [127.0.0.1])
	by mwinf5128.me-wanadoo.net (SMTP Server) with ESMTP id 4F8091C0596C
	for <wfr35c6e058000000055a6c700b@back23-mail02-03.me-wanadoo.net>; Sun, 21 Jan 2007 23:58:00 +0100 (CET)
Received: from smtp23.orange.fr (mwinf2314 [10.232.4.42])
	by mwinf5128.me-wanadoo.net (SMTP Server) with ESMTP id 249291C0EA34
	for <wfr35c6e058000000055a6c700b@0100.back23-mail02-03.me-wanadoo.net>; Sun, 21 Jan 2007 23:50:17 +0100 (CET)
Received: from me-wanadoo.net (localhost [127.0.0.1])
	by mwinf2314.orange.fr (SMTP Server) with ESMTP id 184F62400098
	for <wfr35c6e058000000055a6c700b@0100.back23-mail02-03.me-wanadoo.net>; Sun, 21 Jan 2007 23:50:17 +0100 (CET)
Received: from lists-outbound.sourceforge.net (lists-outbound.sourceforge.net [66.35.250.225])
	by mwinf2314.orange.fr (SMTP Server) with ESMTP id B25AD2400094;
	Sun, 21 Jan 2007 23:50:16 +0100 (CET)
X-ME-UUID: 20070121225016730.B25AD2400094@mwinf2314.orange.fr
Received: from sc8-sf-list2-new.sourceforge.net (sc8-sf-list2-new-b.sourceforge.net [10.3.1.94])
	by sc8-sf-spam2.sourceforge.net (Postfix) with ESMTP
	id 81FB612470; Sun, 21 Jan 2007 14:50:10 -0800 (PST)
Received: from sc8-sf-mx2-b.sourceforge.net ([10.3.1.92]
	helo=mail.sourceforge.net)
	by sc8-sf-list2-new.sourceforge.net with esmtp (Exim 4.43)
	id 1H8lDY-00084P-Lu
	for fink-devel@lists.sourceforge.net; Sun, 21 Jan 2007 14:31:16 -0800
Received: from zakopane.cs.ubc.ca ([198.162.52.68] helo=mail.quuxuum.com)
	by mail.sourceforge.net with esmtp (Exim 4.44) id 1H8lDZ-0001Gb-9P
	for fink-devel@lists.sourceforge.net; Sun, 21 Jan 2007 14:31:18 -0800
Received: from zakopane.cs.ubc.ca (localhost.localdomain [127.0.0.1])
	by mail.quuxuum.com (Postfix) with ESMTP id 069AE5C313
	for <fink-devel@lists.sourceforge.net>;
	Sun, 21 Jan 2007 14:31:04 -0800 (PST)
Received: by zakopane.cs.ubc.ca (tmda-sendmail, from uid 103);
	Sun, 21 Jan 2007 14:31:03 -0800 (PST)
Received: from xanadu.kublai.com (xanadu.kublai.com
	[IPv6:2001:5c0:89ec:1:1::2])
	by mail.quuxuum.com (Postfix) with ESMTP id 72DFF5C1EC
	for <fink-devel@lists.sourceforge.net>;
	Sun, 21 Jan 2007 14:31:03 -0800 (PST)
Date: Sun, 21 Jan 2007 14:31:03 -0800
To: fink-devel@lists.sourceforge.net
Message-ID: <20070121223103.GA13657@xanadu.kublai.com>
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="SLDf9lqlvOQaIe6s"
Content-Disposition: inline
X-Operating-System: Darwin 8.8.0 Power Macintosh
User-Agent: Mutt/1.5.13 (2007-01-09)
From: Brendan Cully <brendan@kublai.com>
X-Delivery-Agent: TMDA/1.0.3 (Seattle Slew)
X-Spam-Score: 1.0 (+)
X-Spam-Report: Spam Filtering performed by sourceforge.net.
	See http://spamassassin.org/tag/ for more details.
	Report problems to
	http://sf.net/tracker/?func=add&group_id=1&atid=200001
	1.0 FORGED_RCVD_HELO       Received: contains a forged HELO
X-Mailman-Approved-At: Sun, 21 Jan 2007 14:49:52 -0800
Subject: [Fink-devel] [PATCH] make fink commander provide default feedback
X-BeenThere: fink-devel@lists.sourceforge.net
X-Mailman-Version: 2.1.8
Precedence: list
List-Id: Discussion for Fink developers and package maintainers
	<fink-devel.lists.sourceforge.net>
List-Unsubscribe: <https://lists.sourceforge.net/lists/listinfo/fink-devel>,
	<mailto:fink-devel-request@lists.sourceforge.net?subject=unsubscribe>
List-Archive: <http://sourceforge.net/mailarchive/forum.php?forum=fink-devel>
List-Post: <mailto:fink-devel@lists.sourceforge.net>
List-Help: <mailto:fink-devel-request@lists.sourceforge.net?subject=help>
List-Subscribe: <https://lists.sourceforge.net/lists/listinfo/fink-devel>,
	<mailto:fink-devel-request@lists.sourceforge.net?subject=subscribe>
Sender: fink-devel-bounces@lists.sourceforge.net
Errors-To: fink-devel-bounces@lists.sourceforge.net
X-me-spamlevel: not-spam
X-me-spamrating: 0.021858


--SLDf9lqlvOQaIe6s
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline

Hi all,

Here's a patch to fink commander to make it indicate positive or
negative feedback from the feedback buttons.

---------------------------------------------------------------------------------------
Orange vous informe que cet  e-mail a ete controle par l'anti-virus mail. 
Aucun virus connu a ce jour par nos services n'a ete detecte.


--SLDf9lqlvOQaIe6s
Content-Type: text/plain; charset=us-ascii
Content-Disposition: attachment; filename="finkcommander-feedback.patch"

diff -r a9acf33ff2ff FinkController.m
--- a/FinkController.m	Mon Jan 22 05:43:23 2007 +0000
+++ b/FinkController.m	Sun Jan 21 14:28:55 2007 -0800
@@ -744,12 +744,20 @@ enum {
 {
     NSEnumerator *e = [[tableView selectedPackageArray] objectEnumerator];
     NSString *sig = [installationInfo formattedEmailSig];
+    NSString *feedbackMessage;
     FinkPackage *pkg;
     NSMutableArray *pkgNames = [NSMutableArray arrayWithCapacity:5];
 
     if (!packageInfo){
 		packageInfo = [[FinkPackageInfo alloc] init];
     }
+
+    if (typeOfFeedback == POSITIVE)
+        feedbackMessage = @"Feedback: works for me";
+    else if (typeOfFeedback == NEGATIVE)
+        feedbackMessage = @"Feedback: not good";
+    else
+        feedbackMessage = @"Feedback: ???";
 
     [packageInfo setEmailSig:sig];
     while (nil != (pkg = [e nextObject])){
@@ -765,7 +773,8 @@ enum {
 			! [[pkg installed] isEqualToString:[pkg stable]]){
 			[pkgNames addObject:[pkg name]];
 		}
-		[[NSWorkspace sharedWorkspace] openURL:[packageInfo mailURLForPackage:pkg]];
+        [[NSWorkspace sharedWorkspace] openURL:[packageInfo mailURLForPackage:pkg
+                                                                     withBody:feedbackMessage]];
     }
 	if (typeOfFeedback == POSITIVE && [pkgNames count] > 0){
 		NSString *msg = [pkgNames count] > 1 ? 
diff -r a9acf33ff2ff FinkPackageInfo.h
--- a/FinkPackageInfo.h	Mon Jan 22 05:43:23 2007 +0000
+++ b/FinkPackageInfo.h	Sun Jan 21 14:28:55 2007 -0800
@@ -51,6 +51,6 @@
 
 -(void)setEmailSig:(NSString *)s;
 
--(NSURL *)mailURLForPackage:(FinkPackage *)pkg;
+-(NSURL *)mailURLForPackage:(FinkPackage *)pkg withBody:(NSString *)body;
 
 @end
diff -r a9acf33ff2ff FinkPackageInfo.m
--- a/FinkPackageInfo.m	Mon Jan 22 05:43:23 2007 +0000
+++ b/FinkPackageInfo.m	Sun Jan 21 14:28:55 2007 -0800
@@ -65,11 +65,11 @@
 
 /* Used to set URL attribute for email addresses displayed by Package Inspector and
 	in FinkController's emailMaintainer method */
--(NSURL *)mailURLForPackage:(FinkPackage *)pkg
+-(NSURL *)mailURLForPackage:(FinkPackage *)pkg withBody:(NSString *)body
 { 
 	return [[NSString stringWithFormat: 
-						@"mailto:%@?subject=%@-%@&body=\n\n%@", 
-						[pkg email], [pkg name], [pkg version], emailSig]
+						@"mailto:%@?subject=%@-%@&body=%@\n\n%@", 
+						[pkg email], [pkg name], [pkg version], body, emailSig]
 				URLByAddingPercentEscapesToString];
 }
 
@@ -154,7 +154,7 @@
 		
 	//look for e-mail url and if found turn it into an active link
 	if ([[p email] length] > 0){
-		NSURL *murl = [self mailURLForPackage:p];
+            NSURL *murl = [self mailURLForPackage:p withBody:@""];
 		r = [[description string] rangeOfString:[p email]];
 		[description addAttributes:urlAttributes range:r];
 		[description addAttribute:NSLinkAttributeName

--SLDf9lqlvOQaIe6s
Content-Type: text/plain; charset="us-ascii"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Content-Disposition: inline

-------------------------------------------------------------------------
Take Surveys. Earn Cash. Influence the Future of IT
Join SourceForge.net's Techsay panel and you'll get the chance to share your
opinions on IT & business topics through brief surveys - and earn cash
http://www.techsay.com/default.php?page=join.php&p=sourceforge&CID=DEVDEV
--SLDf9lqlvOQaIe6s
Content-Type: text/plain; charset="us-ascii"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Content-Disposition: inline

_______________________________________________
Fink-devel mailing list
Fink-devel@lists.sourceforge.net
https://lists.sourceforge.net/lists/listinfo/fink-devel

--SLDf9lqlvOQaIe6s--




